backend/app/controllers/spree/admin/products_controller.rb

```diff
@@ -30,6 +30,11 @@ def update
             param_attrs[:option_value_ids] = param_attrs[:option_value_ids].split(',')
           end
         end
+        if updating_variant_image_rules?
```

Comment:
I think we should not combine the rulesets.  We will get no significant performance benefit out of doing that, it makes for a more confusing data model, imo, and after chatting with Richard we already have some differences between the two that will make the logic uglier. If we have shared behavior between the two then let's make some modules or helpers to avoid excessive repetition. But I think we'll a saner data model and cleaner code by keeping them separate.